components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImpl.java

```diff
@@ -221,6 +231,137 @@ public void deleteBrandingPreference(String type, String name, String locale)
         }
     }
 
+    @Override
+    public CustomText addCustomText(CustomText customText) throws BrandingPreferenceMgtException {
+
+        String resourceName = getResourceNameForCustomText(customText.getScreen(), customText.getLocale());
+        String tenantDomain = getTenantDomain();
+        // Check whether a custom text resource already exists with the same name in the particular tenant to be added.
+        if (isResourceExists(CUSTOM_TEXT_RESOURCE_TYPE, resourceName)) {
+            throw handleClientException(ERROR_CODE_CUSTOM_TEXT_ALREADY_EXISTS, tenantDomain);
+        }
+        String preferencesJSON = generatePreferencesJSONFromPreference(customText.getPreference());
+        if (!BrandingPreferenceMgtUtils.isValidJSONString(preferencesJSON)) {
+            throw handleClientException(ERROR_CODE_INVALID_CUSTOM_TEXT_PREFERENCE, tenantDomain);
+        }
+
+        try {
+            InputStream inputStream = new ByteArrayInputStream(preferencesJSON.getBytes(StandardCharsets.UTF_8));
```

Comment:
don't we need to close this input stream at the end?
Better to you try with resource option

Reply:
Adddressed with https://github.com/wso2-extensions/identity-branding-preference-management/commit/fdd3486cfdc51558c723795cf2fb75b6faafcdc5